dockerfile

```diff
@@ -0,0 +1,112 @@
+# Dockerfile for Embedded-Sharepoint
+FROM ubuntu:22.04
+
+# non-interactive mode for apt-get
+ENV DEBIAN_FRONTEND=noninteractive
+
+# Use bash for subsequent RUN commands
+SHELL ["/bin/bash", "-c"]
+
+# ------------------------------------------------------------
+# Update system packages and install base dependencies
+# ------------------------------------------------------------
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    vim \
+    wget \
+    build-essential \
+    gdb-multiarch \
+    openocd \
+    stlink-tools \
+    software-properties-common \
+    libncurses-dev \
+    python3-pip \
+    git \
+    gnupg \
+    dirmngr \
+    bear \
+    ca-certificates \
+    && rm -rf /var/lib/apt/lists/*
+
+# ------------------------------------------------------------
+# Python 3.10
+# ------------------------------------------------------------
+RUN add-apt-repository ppa:deadsnakes/ppa -y && apt-get update
+
+# Install Python 3.10 (system default on jammy) and dev headers
+RUN apt-get install -y --no-install-recommends \
+    python3.10 python3.10-dev python3-pip \
+    && rm -rf /var/lib/apt/lists/*
+
+# Make python3.10 default
+RUN update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.10 1
+
+# ------------------------------------------------------------
+# Arm None EABI Toolchain
+# ------------------------------------------------------------
+ARG ARM_GNU_VERSION=12.2
```

Comment:
oh also there's a more recent version than this

Reply:
see #68

Reply:
arm-gnu-toolchain 12.2 -> 14.2